<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1060</width>
    <height>810</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1060</width>
    <height>810</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1060</width>
    <height>810</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Ray Tracing</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="buttonRender">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>650</y>
      <width>241</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Render</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>10</y>
      <width>261</width>
      <height>581</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Options</string>
    </property>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>241</width>
       <height>311</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Camera</string>
     </property>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>30</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Vertical</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>130</y>
        <width>211</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Zoom</string>
      </property>
     </widget>
     <widget class="QSlider" name="sliderZoom">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>211</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>75</number>
      </property>
      <property name="maximum">
       <number>500</number>
      </property>
      <property name="value">
       <number>400</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksBelow</enum>
      </property>
      <property name="tickInterval">
       <number>10</number>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>221</width>
        <height>121</height>
       </rect>
      </property>
      <property name="title">
       <string>Look At</string>
      </property>
      <widget class="QDoubleSpinBox" name="cameraLookX">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>30</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
        </font>
       </property>
       <property name="minimum">
        <double>-10000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10000.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="cameraLookY">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>60</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
        </font>
       </property>
       <property name="minimum">
        <double>-10000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10000.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="cameraLookZ">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>90</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
        </font>
       </property>
       <property name="minimum">
        <double>-10000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10000.000000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>16</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>X</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>16</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Y</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>16</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Z</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonCenter">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>30</y>
         <width>93</width>
         <height>81</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Center</string>
       </property>
      </widget>
     </widget>
     <widget class="QDial" name="dialVertical">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>50</y>
        <width>81</width>
        <height>81</height>
       </rect>
      </property>
      <property name="minimum">
       <number>-90</number>
      </property>
      <property name="maximum">
       <number>269</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="wrapping">
       <bool>true</bool>
      </property>
      <property name="notchTarget">
       <double>15.000000000000000</double>
      </property>
      <property name="notchesVisible">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QDial" name="dialHorizontal">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>50</y>
        <width>81</width>
        <height>81</height>
       </rect>
      </property>
      <property name="minimum">
       <number>-179</number>
      </property>
      <property name="maximum">
       <number>180</number>
      </property>
      <property name="value">
       <number>180</number>
      </property>
      <property name="sliderPosition">
       <number>180</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="invertedControls">
       <bool>false</bool>
      </property>
      <property name="wrapping">
       <bool>true</bool>
      </property>
      <property name="notchTarget">
       <double>15.000000000000000</double>
      </property>
      <property name="notchesVisible">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>30</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Horizontal</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QPushButton" name="buttonOptions">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>350</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Render Options...</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBoxKd">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>520</y>
       <width>231</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Use kD-Tree</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkboxPartial">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>440</y>
       <width>231</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Use Mosaic Rendering</string>
     </property>
    </widget>
    <widget class="QPushButton" name="buttonKd">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>470</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Generate kD-Tree</string>
     </property>
    </widget>
    <widget class="QPushButton" name="buttonLights">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>400</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lights...</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBoxViewKd">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>551</y>
       <width>231</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string>View kD-Tree</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="buttonSave">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>700</y>
      <width>241</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Save Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonLoad">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>600</y>
      <width>241</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Load File</string>
    </property>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>750</y>
      <width>251</width>
      <height>21</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="labelImage">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>768</width>
      <height>768</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1060</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>buttonLoad</tabstop>
  <tabstop>buttonRender</tabstop>
  <tabstop>buttonSave</tabstop>
  <tabstop>dialHorizontal</tabstop>
  <tabstop>dialVertical</tabstop>
  <tabstop>sliderZoom</tabstop>
  <tabstop>cameraLookX</tabstop>
  <tabstop>cameraLookY</tabstop>
  <tabstop>cameraLookZ</tabstop>
  <tabstop>buttonCenter</tabstop>
  <tabstop>buttonOptions</tabstop>
  <tabstop>buttonLights</tabstop>
  <tabstop>checkboxPartial</tabstop>
  <tabstop>buttonKd</tabstop>
  <tabstop>checkBoxKd</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
